project/Build.scala

```diff
@@ -2444,6 +2444,12 @@ object Build {
 
       fork in Test := true,
       javaOptions in Test += "-Xmx3G",
+      javaOptions in Test += {
+        // Use maximum 8 threads in partest (avoid saturating the memory on machines with lots of processors)
+        val availableProcs = java.lang.Runtime.getRuntime().availableProcessors()
+        val numThreads = if (availableProcs < 1) 1 else if (availableProcs > 8) 8 else availableProcs
```

Comment:
```scala
val numThreads = Math.max(1, Math.min(8, availableProcs))
```

?

Reply:
That works too, though I find it less readable. Especially the `Math.max(1, ...)` bit.

Reply:
I guess it is too late now, but I think `1 max availableProcs min 8` would be a readable alternative (and lower performance compared to `Math.max` should not be an issue here). I guess you write performance sensitive code too much. 😉 Repeating both 1 and 8 in the `if` version certainly does not look nice to me.

Reply:
I read "1 max availableProcs min 8" as "1 but at most `availableProcs` and at least 8". 🤷‍♂️ 
I never liked the infix `min` and `max`. I think I will forever read them the wrong way.